src/enzyme_ad/jax/Passes/AffineCFG.cpp

```diff
@@ -3666,6 +3678,28 @@ struct AddAddCstEnd : public OpRewritePattern<arith::AddIOp> {
   }
 };
 
+static void greedilyParallelize(Operation *op) {
+  // Whether to parallelize reductions.
+  constexpr bool parallelReductions = true;
+
+  // The walker proceeds in pre-order to process the outer loops first
+  // and control the number of outer parallel loops.
+  std::vector<ParallelizationCandidate> parallelizableLoops;
+  op->walk<WalkOrder::PreOrder>([&](AffineForOp loop) {
```

Comment:
we may want to run this before the affinecfg stuff [or ideally as a pattern], since the parallel fusion, and index rewrite is a pattern [otherwise run before parallelize]

Reply:
it's not initially written is a pattern, so I suspect it may mess up the internal state of the rewriter; let me see what I can do